draft-ietf-quic-load-balancers.md

```diff
@@ -324,6 +308,78 @@ the packet. The number of servers is usually consistent on the time scale of a
 QUIC connection handshake. Another might simply hash the address/port 4-tuple.
 See also {{version-invariance}}.
 
+Note that any information used by the Fallback Algorithm might not be repeated
+from one QUIC packet to another. The 4-tuple is assumed to remain stable during
+the handshake, but the server can change the destination connection ID after the
+client's first flight.
+
+Therefore, if the fallback algorithm uses information other than the 4-tuple in
+its routing decision, the load balancer SHOULD store the routing decision for
+that 4-tuple in order to route future packets unless doing so would exceed
+resource limits. It can free these records when the 4-tuple uses a routable CID.
+This state allows the load balancer to successfully route other packets from
+the client's first flight, as well as subsequent flights if the server provides
+an unroutable CID for the reasons described in {{config-failover}}.
+
+Even if the fallback algorithm solely uses the 4-tuple, it MAY store the
+algorithm result in case changes to the server pool change fallback algorithm
+output.
+
```

Comment:
I am not an LB implementer, but aren't you describing some kind of race condition? Once the server has received the first flight, it will provide its source CID to the client. The next packets from the client will use that as a DCID, and it will be routable.

Reply:
Again, all of this is covering the case when the server does not have a config and is using 0b111.

Reply:
OK.